client/server/middleware/logger.js

```diff
@@ -1,43 +1,54 @@
+/**
+ * External dependencies
+ */
+import uaParser from 'ua-parser-js';
+
 /**
  * Internal dependencies
  */
 import { getLogger } from 'server/lib/logger';
+import config from 'config';
 
 const NS_TO_MS = 1e-6;
 
+const parseUA = ( rawUA ) => {
+	const parsedUA = uaParser( rawUA );
+	if ( parsedUA.browser.name && parsedUA.browser.version ) {
+		const version = parsedUA.browser.version.split( '.' )[ 0 ];
+		return `${ parsedUA.browser.name } ${ version }`;
+	}
+	return rawUA;
+};
+
 const logRequest = ( req, res, options ) => {
-	const { requestStart } = options;
+	const { requestStart, env, version } = options;
 
-	let fields = {
+	const fields = {
 		method: req.method,
 		status: res.statusCode,
 		length: res.get( 'content-length' ),
 		url: req.originalUrl,
 		duration: Number(
 			( Number( process.hrtime.bigint() - requestStart ) * NS_TO_MS ).toFixed( 3 )
 		),
+		httpVersion: req.httpVersion,
+		version,
+		env,
+		userAgent: parseUA( req.get( 'user-agent' ) ),
```

Comment:
Previously we used to supply the user agent in a raw format and it included a lot more information than just the browser and its version. While this information was a bit more difficult to analyze, it was also a bit more informative.

In that regard, do you think that the simplified user agent data with the browser name and version makes sense to added as another field here, allowing us to keep the original raw `userAgent` field? Perhaps something like `userAgentBrowser` or similar?

Reply:
Makes sense, I'll add it.